src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -1014,21 +762,17 @@ internal void FreePageOnCommit(long pageNumber)
 
         internal void DiscardScratchModificationOn(long pageNumber)
         {
-            if (_scratchPagesTable.TryGetValue(pageNumber, out var scratchPage))
+            var scratchPagesInUse = _env.WriteTransactionPool.ScratchPagesInUse;
+            if (scratchPagesInUse.Remove(pageNumber, out var scratchPage) 
```

Comment:
Does it call Remove on ImmutableDictionary?

Reply:
On the `Builder` not the dictionary. This is creating the state for the next transaction (T+1)

Reply:
No, note that this `ImmutableDictionary.Builder`